mlflow/pyfunc/utils/data_validation.py

```diff
@@ -0,0 +1,122 @@
+import inspect
+import warnings
+from functools import lru_cache, wraps
+from typing import Any, Optional
+
+from mlflow.models.signature import (
+    _extract_type_hints,
+    _is_context_in_predict_function_signature,
+    _TypeHints,
+)
+from mlflow.types.type_hints import (
+    _convert_data_to_type_hint,
+    _infer_schema_from_type_hint,
+    _signature_cannot_be_inferred_from_type_hint,
+    _validate_example_against_type_hint,
+)
+from mlflow.utils.annotations import filter_user_warnings_once
+
+
+def _wrap_predict_with_pyfunc(func, type_hint):
```

Comment:
nit: shall we define this protected method below "pyfunc"?

Reply:
ok